clustering/service/src/main/java/org/wildfly/clustering/service/ServiceNameRegistry.java

```diff
@@ -28,7 +28,7 @@
  * Registry of services names for a set of requirements.
  * @author Paul Ferraro
  */
-@Deprecated
+@Deprecated(forRemoval = true)
```

Comment:
Tangential to the `forRemoval` – it is strongly recommended that the reason for deprecating a program element be explained in the documentation, using the @deprecated javadoc tag. The documentation should also suggest and link to a recommended replacement API, if applicable. A replacement API often has subtly different semantics, so such issues should be discussed as well.

Perhaps the ` * @deprecated To be removed without replacement` would suffice.

Reply:
Same applies to the enum members below.